### PR TITLE
Ports aiming from NSV13 (ranged holdup anyone?)

### PR DESCRIPTION
**Pull request by maxion12345**
# General Documentation

### Intent of your Pull Request

Implements ranged aiming. Similar to holdups, but you can do it from a range. You can give commands to the person like "HANDS UP" as well. Initiated by pointing at someone with a gun. Also works with banana. Should give some more RP fun-ness to security, as aside from the usual "80% sure this dude valid, gonna stun him with everyone I have and quickly drag into a security office to strip-search". This is not a complete replacement of the holdup system, as the holdup system is still useful for hostage-taking where the damage multiplication from that can be useful in ensuring your threats are taken seriously. Important to note that, unlike holdup, you'll need to press the button to fire yourself. So you'll be alerted when they try something, but its up to your reaction time to take the shot!

To-do
Exclude particle rifle
Maybe maybe remove surrender button. Or put it in chat somehow?
### Why is this change good for the game?
This will enable some more RP-based interactions with firearms, that aren't limited to just shooting. I've always thought holdups were limited in their scope due to the fact you need to actually smack the person with the weapon to initiate a holdup. Not to mention stopping to type something like "FREEZE" usually ends up in you being shot or the perp just running while you type. My hope is the loud indicator and obvious "surrender" button will entice people to surrender more than run from security, or a traitor taking a hostage, or clown with banana.
# Wiki Documentation

### Briefly describe your PR and the impacts of it, in layman's terms. 
Implements a ranged "aiming" system, where you can point with a gun at something to aim at them. This gives you an options wheel of voice commands, and the perp gets a "surrender" button. 

### What should players be aware of when it comes to the changes your PR is implementing?
Point at someone with a gun to initiate aiming. Firing when aiming increases your bullet speed to make it harder to dodge, so move fast if you want to avoid it! Dropping items, equipping items, and moving gives an alert to the person aiming. It won't auto-fire though! 
### What general grouping does this PR fall under? 
General tweaks

### Are there any aspects of the PR that you would like us not to mention on the Wiki?
yeah don't mention it adds me as council kthanks
### If there are any numerical values involved in your PR that will be relevant to a player, please note them here. 
Firing while aiming increases bullet speed by about 30%. I don't wanna do math but it seems that much faster.

# Changelog

:cl:  
rscadd: Added aiming! 
rscadd: Point at someone with a gun in your active hand to "aim" it at them. This lets you give various voice commands via an options wheel. 

/:cl:

### CODE REVIEW

**Comment by maxion12345** [comment]
i dont wanna fix the errors right now i'll start on this /later/

**Comment by maxion12345** [comment]
thank you @TheGamerdk

**Comment by Cartlord** [comment]
Please don't, Holdup is already quite powerful IMO and being able to do it at range is just ridiculous.

**Comment by maxion12345** [comment]
> Please don't, Holdup is already quite powerful IMO and being able to do it at range is just ridiculous.

It's not a holdup. You won't be stopped/interrupted and the shooting is not automatic. It really boils down to the framework to give orders to someone at gunpoint, perhaps as a prelude to a holdup?

**Review by nesisknesis — APPROVED** [review]

**Comment by Cartlord** [comment]
Oh good.

File: yogstation/code/modules/jobs/security/aiming.dm [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+		if(isliving(A))
+			aiming.aim(user, A)
+		return
+	. = ..()
+
+//Happens before the actual projectile creation
+/obj/item/gun/before_firing(atom/target,mob/user, aimed)
+	if(aimed)
+		if(chambered?.BB && !istype(src, /obj/item/gun/ballistic/automatic/toy))
+			chambered.BB.stamina = initial(chambered.BB.stamina) += 55
```
Comment (SomeguyManperson) [inline yogstation/code/modules/jobs/security/aiming.dm]: this happens
not a complaint just a statement

**Comment by N3D6** [comment]
currently this is bad because you can literally stunlock people with laser tag guns, even if they're not wearing armor

**Comment by maxion12345** [comment]
> currently this is bad because you can literally stunlock people with laser tag guns, even if they're not wearing armor

Stunlock as in the laser tag gun does stamina damage without the suit on?

**Comment by Marmio64** [comment]
yeah lasergun can stamcrit people, even if the enemy doesnt have laser tag armor.
And, the surrender button needs to be at the top left instead or something maybe, because right now its too easy to aim at someone and then have them accidently press the surrender button thats right next to them and they lose instantly.

File: yogstation/code/modules/jobs/security/aiming.dm (hunk 1 above)
Comment (SomeguyManperson) [reply yogstation/code/modules/jobs/security/aiming.dm]: ok this is a\ctually a massive bruh moment now it needs to not give stamina damage to everything
normal holdup gets away with boosted damage because it's both triggered automatically and requires time with this you can spam it

**Comment by SomeguyManperson** [comment]
add a cooldown for using the voice commands too it is kind of funny to spam them but it will stop being funny and can trigger spam automute

File: yogstation/code/modules/jobs/security/aiming.dm (hunk 1 above)
Comment (maxion12345) [reply yogstation/code/modules/jobs/security/aiming.dm]: Didn't realize it was giving the extra damage. It was under the code for the "make bullet a bit faster" section so I assumed it was something unique to non-lethal projectiles or something. I'll go remove it here in a bit, didn't intend to have any damage buffs in here anyways

**Comment by maxion12345** [comment]
> add a cooldown for using the voice commands too it is kind of funny to spam them but it will stop being funny and can trigger spam automute

I mean if they spam shit they get automated isn't that how it works

I'll see about a cool down if I can

**Comment by N3D6** [comment]
> > currently this is bad because you can literally stunlock people with laser tag guns, even if they're not wearing armor
> 
> Stunlock as in the laser tag gun does stamina damage without the suit on?

yes

**Comment by maxion12345** [comment]
removed stamina damage thing, gonna work on a cooldown to the voice command now

**Comment by maxion12345** [comment]
voice command cooldown is done

**Comment by maxion12345** [comment]
yeah so slight issue, the fuckin...particle rifle shoots instantly with this

gonna have to exclude it somehow

and move the surrender button somehow

**Comment by Cartlord** [comment]
One possible issue is that you nede to look in chat to know what you're being held up with - had one Botanist actually mug someone using a banana, because the other person thought he had a gun.

**Comment by maxion12345** [comment]
Been working recently so this hasn't been getting the attention it needs. Anyways, in regards to checking chat to see what your being held up with, it would be faster to just examine the person to check what they are holding. It'll be easier on the eyes due to aiming putting a small paragraph of text in there.

**Comment by enkidienne** [comment]
Having played with it, I don't like it. Its clunky and there is no way I'm gonna use it in a normal situation over my personal speed and reflex.

**Comment by maxion12345** [comment]
It's meant more to appeal to the "mrp" part of our game. You aren't required to use it at all, and using it offers minimal benefits at best. 

I've been swamped with schoolwork recently, so this probably won't get an update until a few more days.

**Comment by JamieD1** [comment]
Fun but doesn't really work in practice very well.
